### PR TITLE
OOP Exercise

### PR DESCRIPTION
**Pull request by crisgahur**
Uploading exercise of ticket [BACK-4](https://vulky.atlassian.net/browse/BACK-4).

This is a OOP exercise developed in Java. In this proyect you will find these concepts applied:

- Class.
- Methods
- Attributes
- Inheritance
- Extends.
- Implements
- Polymorphism.
- Interface
- Objects as parameter
- Model class
- Encapsulation

### CODE REVIEW

**Comment by MauroDecodificado** [comment]
Title is not correct "POO Exercise 1".

Also please add a good description for this PR

File: README.md [hunk 1]
```diff
@@ -1 +1,36 @@
-# OOPExercise
\ No newline at end of file
+# POOExerciseONE
```
Comment (MauroDecodificado) [inline README.md:1]: Awesome! 🚀

File: src/main/java/com/mycompany/exerciseone/Pedro.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/**
+ *
+ * @author Cristian Gallego
+ */
+public class Pedro extends Person {
+    
+    public void  playPiano(String song){
+     System.out.println(song);
+    };
+
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/Pedro.java]: Please delete empty space

File: src/main/java/com/mycompany/exerciseone/Person.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ *
+ * @author Cristian Gallego
+ */
+public class Person implements InterfaceActionPerson {
+    
+    public PersonModel globalVariableModelPerson;
+    
+    public void setAttributes(PersonModel localVariableModelPerson ){
+
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/Person.java]: Please delete empty space

File: src/main/java/com/mycompany/exerciseone/PersonModel.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/PersonModel.java:36]: Please delete empty space

File: src/main/java/com/mycompany/exerciseone/PersonModel.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+    
+    
+       public PersonModel(){};
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/PersonModel.java]: Indent code

File: src/main/java/com/mycompany/exerciseone/PersonModel.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    
+       public PersonModel (String hairColor, String name, int age){
+        this.hairColor = hairColor;
+        this.name = name;
+        this.age = age;
+    }
+
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/PersonModel.java:44]: Please delete empty space

File: src/main/java/com/mycompany/exerciseone/Person.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    @Override
+    public void jump() {
+    System.out.println("Jump twice");
+    }
+
+    @Override
+    public void scream() {
+           System.out.println("Hellooo");
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/Person.java]: Indent code

File: src/main/java/com/mycompany/exerciseone/Pepita.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ *
+ * @author Cristian Gallego
+ */
+public class Pepita extends Person{
+
+    public String playFlute(String song) {
+        return song; 
+    }
+;
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/Pepita.java]: What is this?

File: src/main/java/com/mycompany/exerciseone/ExerciseOne.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        pedrito.walk();
+        
+         System.out.println("");
+        
+        Pepita pepera = new Pepita();
+        PersonModel modelPepita = new PersonModel("red", "Pepita", 23);
+        pepera.setAttributes(modelPepita);
+        pepera.walk();
+        System.out.println(pepera.playFlute("Piripipipi"));
+
```
Comment (MauroDecodificado) [inline src/main/java/com/mycompany/exerciseone/ExerciseOne.java]: Delete empty space

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Check comments

File: src/main/java/com/mycompany/exerciseone/Pedro.java (hunk 2 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/Pedro.java]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/Person.java (hunk 3 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/Person.java]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/PersonModel.java (hunk 4 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/PersonModel.java:36]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/PersonModel.java (hunk 5 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/PersonModel.java]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/PersonModel.java (hunk 6 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/PersonModel.java:44]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/Person.java (hunk 7 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/Person.java]: Ok, done.

File: src/main/java/com/mycompany/exerciseone/ExerciseOne.java (hunk 9 above)
Comment (crisgahur) [reply src/main/java/com/mycompany/exerciseone/ExerciseOne.java]: Ok, done

File: README.md (hunk 1 above)
Comment (crisgahur) [reply README.md:1]: Thankss

**Review by MauroDecodificado — APPROVED** [review]

**Comment by MauroDecodificado** [comment]
PR approved and can be merged!

**Pull request merged**